CONTRIBUTING.md

```diff
@@ -0,0 +1,114 @@
+# Contributing guidelines
+
+Welcome to [TheAlgorithms/Nim](https://github.com/TheAlgorithms/Nim)! Before sending your pull requests, make sure that you __read the whole guidelines__. If you have any doubt on the contributing guide, please feel free to [state it clearly in an issue](https://github.com/TheAlgorithms/Nim/issues/new) or ask the community in [Discord](https://discord.gg/c7MnfGFGa6).
+
+## Contributing
+
+We are very happy that you are considering implementing algorithms and data structures for others!
+
+ This repository is referenced and used by learners from all over the globe. Being one of our contributors, you agree and confirm that:
+
+- You did your work - no plagiarism allowed
+  - Any plagiarized work will not be merged.
+- Your work will be distributed under [MIT License](LICENSE.md) once your pull request is merged
+- Your submitted work fulfills our styles and standards
+
+__New implementation__ is welcome! For example, new solutions for a problem, different representations for a graph data structure or algorithm designs with different complexity but __identical implementation__ of an existing implementation is not allowed. Please check whether the solution is already implemented or not before submitting your pull request.
```

Comment:
```suggestion
**New implementations** are welcome! For example, new solutions for a problem, different representations for a graph data structure, or algorithm designs with different complexity but **identical implementation** of an existing implementation are not allowed. Please check whether the solution is already implemented or not before submitting your pull request.
```

Reply:
Identical implementation is singular if I am not mistaken, so *is* not allowed looks correct to me.